### PR TITLE
fix: add reflection configurations for com.google.rpc classes

### PR DESCRIPTION
**Pull request by mpeddada1**
For test failures in https://github.com/googleapis/java-spanner/pull/2615

This PR adds native image metadata for the classes that are being invoked reflectively. The call to `ProtoUtils.keyForProto` in `SpannerInterceptorProvider` calls `GeneratedMessageV3#getMethodOrDie` down the line which then makes use of reflection to invoke methods in the `google.rpc` classes.  

https://github.com/protocolbuffers/protobuf/blob/2bea49218587ddb9cb2f76b75618ba70b1e11fa5/java/core/src/main/java/com/google/protobuf/GeneratedMessageV3.java#L1999-L2008

Stacktrace from IntelliJ debugger:
Screenshot 2023-09-11 at 5 42 48 PM

### CODE REVIEW

**Comment by blakeli0** [comment]
For future improvement, if we don't worry too much about extra reflection configs, we can generate a common json for every common proto message in [java-common-protos](https://github.com/googleapis/sdk-platform-java/tree/main/java-common-protos/proto-google-common-protos). Otherwise we may have to add these files manually for every handwritten library as we don't know what exactly common proto messages they are using.

**Review by olavloite — APPROVED** [review]

File: google-cloud-spanner/src/main/resources/META-INF/native-image/com.google.cloud.spanner/reflect-config.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
+[ {
+  "name": "com.google.rpc.LocalizedMessage",
+  "queryAllDeclaredConstructors": true,
+  "queryAllPublicConstructors": true,
+  "queryAllDeclaredMethods": true,
+  "allPublicMethods": true,
+  "allDeclaredClasses": true,
+  "allPublicClasses": true
+},
```
[the remark this answers is not in the document]
Comment (olavloite) [reply google-cloud-spanner/src/main/resources/META-INF/native-image/com.google.cloud.spanner/reflect-config.json:9]: (Let's do this in a separate PR, and merge this now, so we can proceed with the releases)

[the remark this answers is not in the document]
Comment (mpeddada1) [reply google-cloud-spanner/src/main/resources/META-INF/native-image/com.google.cloud.spanner/reflect-config.json:9]: Applied change in follow-up PR #2621. Thanks.

**Comment by mpeddada1** [comment]
> For future improvement, if we don't worry too much about extra reflection configs, we can generate a common json for every common proto message in [java-common-protos](https://github.com/googleapis/sdk-platform-java/tree/main/java-common-protos/proto-google-common-protos). Otherwise we may have to add these files manually for every handwritten library as we don't know what exactly common proto messages they are using.

Agreed! This is something that we should we consider + discuss further. Additionally, we rely quite a bit tests to detect native image incompatibilities so if a handwritten library adds new code that explicitly uses `com.google.rpc` classes but doesn't test for that code path (or uses canary tests) then we may risk pushing incompatible code.

**Pull request merged**